doc/modules/feature_extraction.rst

```diff
@@ -101,6 +101,68 @@ memory the ``DictVectorizer`` class uses a ``scipy.sparse`` matrix by
 default instead of a ``numpy.ndarray``.
 
 
+.. _column_transformer:
+
+ColumnTransformer for heterogeneous data
+========================================
+
+.. warning::
+
+    The :class:`experimental.ColumnTransformer <sklearn.experimental.ColumnTransformer>`
+    is experimental and the API is subject to change.
+
+Many datasets contain features of different types, say text, floats and dates,
+where each type of feature requires separate preprocessing or feature extraction steps.
+Often it is easiest to preprocess data before applying scikit-learn methods,
+for example using :ref:`pandas <http://pandas.pydata.org/>`__.
+The :class:`~sklearn.experimental.ColumnTransformer` is a convenient way to perform heterogeneous
+preprocessing on data columns within a scikit-learn pipeline (for example,
+when you want to adjust preprocessing parameters within a grid search).
+The :class:`~sklearn.experimental.ColumnTransformer` works on (sparse) arrays and pandas dataframes.
+
+To each column, a different transformation can be applied, such as
+preprocessing or a specific feature extraction method::
+
+  >>> import pandas as pd
+  >>> X = pd.DataFrame(
+  ...     {'city': ['London', 'London', 'Paris', 'New York'],
+  ...      'title': ["His Last Bow", "How Watson Learned the Trick", "A Moveable Feast", "The Great Gatsby"]})
+
+For this data, we might want to apply a :class:`preprocessing.OneHotEncoder` to the
```

Comment:
Maybe in this description just say we want to encode city as categorical variable, because that's what were doing below, but not using OneHotEncoder